### PR TITLE
 Fix NVFP4-quantized MoE checkpoint support for Step-3.5 Flash

### PR DESCRIPTION
**Pull request by meenchen**
## Purpose

  Fix NVFP4-quantized MoE checkpoint support for Step-3.5 Flash. Three bugs in step3p5.py:

  1. MoE experts loaded as BF16 instead of NVFP4:   FusedMoEBlock and Step3p5MLP had extra is_layer_excluded() checks that incorrectly set quant_config=None for expert layers, bypassing quantization. The checkpoint's hf_quant_config.json exclusion list already handles this — the redundant checks caused double-exclusion. Fix: pass quant_config directly.

  2. Per-expert input_scale never loaded → NaN logits → empty output:   expert_params_mapping in load_weights only matched keys containing "weight" as a substring, so gate_proj.input_scale / up_proj.input_scale / down_proj.input_scale were silently dropped. This left w13_input_scale and w2_input_scale as uninitialized torch.empty(), producing NaN logits and all-zero token IDs on every decode step. Fix: add explicit input_scale entries to expert_params_mapping.

  3. Missing 0-dim scalar guard in weight loading loop:   Fix: add ndim == 0 check before the existing shape[0] == 1 expansion, to handle checkpoints that store scales as bare scalars.

## Test Plan

Tested with an internal NVFP4 checkpoint quantized by

## Test Result

 ✓ Memory: 56 GiB (quantized, vs ~120 GiB unquantized)
 ✓ Throughput: ~110 tok/s (TP=2, 2×RTX PRO 6000 Blackwell)

  Before: all token IDs = 0, empty output, finish_reason: length
  After:  coherent generation, finish_reason: stop

  Prompt: What is the capital of France?
  → The capital of France is Paris.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/step3p5.py [hunk 1]
```diff
@@ -381,7 +384,7 @@ def __init__(
             intermediate_size=config.moe_intermediate_size,
             reduce_results=False,
             renormalize=config.norm_expert_weight,
-            quant_config=quant_config,
+            quant_config=quant_config,  # Pass quant_config correctly
```
Comment (mgoin) [inline vllm/model_executor/models/step3p5.py]: Remove cruft comments?

File: vllm/model_executor/models/step3p5.py [hunk 2]
```diff
@@ -647,6 +650,12 @@ def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]) -> set[str]:
             (".moe.experts.w13_weight", ".moe.gate_proj.weight", "w1"),
             (".moe.experts.w13_weight", ".moe.up_proj.weight", "w3"),
             (".moe.experts.w2_weight", ".moe.down_proj.weight", "w2"),
+            # NVFP4: per-expert input scales — must be loaded explicitly or
+            # w13_input_scale / w2_input_scale stay torch.empty (uninitialized),
+            # causing NaN logits and all-zero token generation.
+            (".moe.experts.w13_input_scale", ".moe.gate_proj.input_scale", "w1"),
+            (".moe.experts.w13_input_scale", ".moe.up_proj.input_scale", "w3"),
+            (".moe.experts.w2_input_scale", ".moe.down_proj.input_scale", "w2"),
```
Comment (mgoin) [inline vllm/model_executor/models/step3p5.py]: How do we handle this for other models without this same fix? It seems it shouldn't be needed

File: vllm/model_executor/models/step3p5.py (hunk 1 above)
Comment (meenchen) [reply vllm/model_executor/models/step3p5.py]: Removed

File: vllm/model_executor/models/step3p5.py (hunk 2 above)
Comment (meenchen) [reply vllm/model_executor/models/step3p5.py]: The need for these entries is specific to this model's checkpoint format. The Step-3.5 checkpoint uses the old packed format where all expert tensors are stored with the same naming convention: `moe.gate_proj.weight (shape [num_experts, out, in])`

**Review by ajrasane — APPROVED** [review]
